src/main/scala/com/wavesplatform/state2/StateWriter.scala

```diff
@@ -93,8 +94,45 @@ class StateWriterImpl(p: StateStorage, synchronizationToken: ReentrantReadWriteL
     }
 
     measureSizeLog("contracts")(blockDiff.txsDiff.contracts) {
-      _.foreach { case (name, (status, account, content)) =>
-        sp().contracts.put(name, (status, account.bytes, content))
+      _.foreach { case (id, (h, txId, contract, _)) =>
+        sp().contracts.put(id, (h, txId, contract.bytes.arr))
+      }
+    }
+
+    measureSizeLog("accountContractIds")(blockDiff.txsDiff.accountContractIds) {
+      _.foreach { case (acc, ctIds) =>
+        val startIdxShift = sp().accountContractsLengths.get(acc.bytes).getOrElse(0)
```

Comment:
what does "shift" do?

Reply:
accountContractIds DB will store all contracts that are registered by this address. Keys to this DB are constructed be addressBytes + count of a contract(shift). For each time you register a contract, shift will increase by 1.